src/coreclr/jit/lower.cpp

```diff
@@ -4891,20 +4891,29 @@ bool Lowering::AreSourcesPossiblyModifiedLocals(GenTree* addr, GenTree* base, Ge
 //    addressing mode and transform them to a GT_LEA
 //
 // Arguments:
-//    use - the use of the address we want to transform
+//    addr - the use of the address we want to transform
 //    isContainable - true if this addressing mode can be contained
-//    targetType - on arm we can use "scale" only for appropriate target type
+//    parent - the node that consumes the given addr (most likely it's an IND)
 //
 // Returns:
 //    true if the address node was changed to a LEA, false otherwise.
 //
-bool Lowering::TryCreateAddrMode(GenTree* addr, bool isContainable, var_types targetType)
+bool Lowering::TryCreateAddrMode(GenTree* addr, bool isContainable, GenTree* parent)
 {
     if (!addr->OperIs(GT_ADD) || addr->gtOverflow())
     {
         return false;
     }
 
+#ifdef TARGET_ARM64
+    if (parent->OperIsIndir() && parent->gtFlags & GTF_IND_VOLATILE)
```

Comment:
Could be `parent->AsIndir()->IsVolatile()`.

Reply:
sure, but don't want to re-run CI for a minor change (if it will be the only requested one) 🙂 will include as part of some other PR